lib/board.rb

```diff
@@ -1,44 +1,58 @@
 class Board 
 attr_reader :cells
+
   def initialize
-    @cells = {
-        "A1" => Cell.new("A1"),
-        "A2" => Cell.new("A2"),
-        "A3" => Cell.new("A3"),
-        "A4" => Cell.new("A4"),
-        "B1" => Cell.new("B1"),
-        "B2" => Cell.new("B2"),
-        "B3" => Cell.new("B3"),
-        "B4" => Cell.new("B4"),
-        "C1" => Cell.new("C1"),
-        "C2" => Cell.new("C2"),
-        "C3" => Cell.new("C3"),
-        "C4" => Cell.new("C4"),
-        "D1" => Cell.new("D1"),
-        "D2" => Cell.new("D2"),
-        "D3" => Cell.new("D3"),
-        "D4" => Cell.new("D4")
-      }
+    @cells = create_cells
+    @letter = ("A".."D").to_a
+    @number = (1..4).to_a
+  end 
+
+  def create_cells 
+    cells = {}
+    ("A".."D").each do |letter|
+      (1..4).each do |number|
+        coordinate = "#{letter}#{number}"
+        cells[coordinate] = Cell.new(coordinate)
+      end
+    end
+      cells
   end
 
   def valid_coordinate?(coordinate)
-    if @cells.keys.include?(coordinate)
-      true
-    else 
-      false
-    end
+    @cells.key?(coordinate)
   end
 
   def valid_placement?(ship, coordinates)
+    return false unless ship.length == coordinates.count
```

Comment:
Good naming convention for the helper method. How about we mark this for future refactoring to see if we can make this code more robust?